src/braket/circuits/circuit.py

```diff
@@ -487,6 +492,77 @@ def add_circuit(
 
         return self
 
+    def add_verbatim_box(
+        self,
+        verbatim_circuit: Circuit,
+        target: QubitSetInput = None,
+        target_mapping: Dict[QubitInput, QubitInput] = None,
+    ) -> Circuit:
+        """
+        Add a verbatim `circuit` to self, that is, ensures that `circuit` is not modified in any way
+        by the compiler.
+
+        Args:
+            verbatim_circuit (Circuit): Circuit to add into self.
+            target (int, Qubit, or iterable of int / Qubit, optional): Target qubits for the
+                supplied circuit. This is a macro over `target_mapping`; `target` is converted to
+                a `target_mapping` by zipping together a sorted `circuit.qubits` and `target`.
+                Default = `None`.
+            target_mapping (dictionary[int or Qubit, int or Qubit], optional): A dictionary of
+                qubit mappings to apply to the qubits of `circuit.instructions`. Key is the qubit
+                to map, and the value is what to change it to. Default = `None`.
+
+        Returns:
+            Circuit: self
+
+        Raises:
+            TypeError: If both `target_mapping` and `target` are supplied.
+            ValueError: If `circuit` has result types attached
+
+        Examples:
+            >>> widget = Circuit().h(0).h(1)
+            >>> circ = Circuit().add_verbatim_box(widget)
+            >>> print(list(circ.instructions))
+            [Instruction('operator': StartVerbatimBlock, 'target': QubitSet([])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(0)])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(1)])),
+             Instruction('operator': EndVerbatimBlock, 'target': QubitSet([]))]
+
+            >>> widget = Circuit().h(0).cnot(0, 1)
+            >>> circ = Circuit().add_verbatim_box(widget, target_mapping={0: 10, 1: 11})
+            >>> print(list(circ.instructions))
+            [Instruction('operator': StartVerbatimBlock, 'target': QubitSet([])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(10)])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(11)])),
+             Instruction('operator': EndVerbatimBlock, 'target': QubitSet([]))]
+
+            >>> widget = Circuit().h(0).cnot(0, 1)
+            >>> circ = Circuit().add_verbatim_box(widget, target=[10, 11])
+            >>> print(list(circ.instructions))
+            [Instruction('operator': StartVerbatimBlock, 'target': QubitSet([])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(10)])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(11)])),
+             Instruction('operator': EndVerbatimBlock, 'target': QubitSet([]))]
+        """
+        if target_mapping and target is not None:
```

Comment:
Do we need to support both target and target_mapping at launch ? Was this a requirement ?

Reply:
Wasn't a requirement, and I can remove this. However, the reason I made this was for consistency with the rest of the `add_` methods